codalab/worker/docker_image_manager.py

```diff
@@ -82,6 +74,29 @@ def stop(self):
             self._cleanup_thread.join()
         logger.info("Stopped docker image manager")
 
+    def _get_cache_use(self):
+        return sum(
+            float(image.attrs['VirtualSize']) for image in self._docker.images.list(self.CACHE_TAG)
+        )
+
+    def _tag_local_images(self):
```

Comment:
@bkgoksel I added a new function `_tag_local_images` for the very first release. This function will tag all the existing images. Then they will be picked up in the cleanup phase.

Reply:
I think this makes too strong an assumption that the worker is responsible for all the docker images on its host machine. This makes sense in the public codalab workers, but in a shared tenancy deployment (like in Slurm or AWS Batch) there might be docker images downloaded by other users/system components that we do not care about/should not mess with. Calling this would wrongly make our worker try to prune those images later on.

I think it's better to manually tag existing images on worker update only for this update (or prune all existing images on public workers and start afresh with a new cache once the update is done).

Reply:
Yes. I agree with you on this. The more that I was thinking is that I think this function can be used to run only during the very first release. Then all the processes will just be automated during the next release. Since we don't need to do this after the first release, once the first release is done, we can remove this function from `start()`. In the future, if there is any change that people need to tag their local images, they can just manually tag their images using this function.

Reply:
Created a github issue to track this https://github.com/codalab/codalab-worksheets/issues/1949